packages/hub/tasks/scheduled-payment-on-chain-execution-waiter.ts

```diff
@@ -0,0 +1,85 @@
+import { inject } from '@cardstack/di';
+import { isBefore, subDays } from 'date-fns';
+
+import { nowUtc } from '../utils/dates';
+
+export default class ScheduledPaymentOnChainExecutionWaiter {
+  prismaManager = inject('prisma-manager', { as: 'prismaManager' });
+  web3 = inject('web3-http', { as: 'web3' });
+  cardpay = inject('cardpay');
+  workerClient = inject('worker-client', { as: 'workerClient' });
+
+  async perform(payload: { scheduledPaymentAttemptId: string }) {
+    let prisma = await this.prismaManager.getClient();
+
+    let paymentAttempt = await prisma.scheduledPaymentAttempt.findFirstOrThrow({
+      where: { id: payload.scheduledPaymentAttemptId },
+    });
+
+    if (paymentAttempt.status != 'inProgress' || !paymentAttempt.transactionHash) {
+      return; // Such attempt should not be waited for
+    }
+
+    let scheduledPaymentModule = await this.cardpay.getSDK('ScheduledPaymentModule', this.web3.getInstance());
+
+    try {
+      await scheduledPaymentModule.executeScheduledPayment(paymentAttempt.transactionHash); // Will wait until mined. If already mined, will return immediately
+      await prisma.scheduledPaymentAttempt.update({
+        where: { id: paymentAttempt.id },
+        data: { status: 'succeeded', endedAt: nowUtc() },
+      });
+    } catch (error: any) {
+      // waitUntilTransactionMined will return "Transaction took too long to complete" in case it wasn't mined in 60 minutes.
+      // In this case we want to restart the task and wait some more. We could throw an error here so that the worker would restart the task, but
+      // due to the exponential-backoff it could take some time before the waiting can start again. That's why we just spawn a new task directly.
+      if (error.message.includes('took too long')) {
```

Comment:
This check can fall apart in case someone changes the error copy in the SDK 😬 I guess a more proper solution would be to wrap the error in the SDK in a new error type, e.g `TimeoutError`